### ISSUE TITLE
Add support for x token in formatter

### ISSUE THREAD

**Issue by jadchaar** [crsmithdev/arrow#709]
Support for parsing integer timestamps with milliseconds and microseconds was added in https://github.com/crsmithdev/arrow/pull/684, but we need to add similar support to the formatter.

**Comment by gramirez093** [comment]
I will look into this!

**Comment by alstomli** [comment]
By using a command like "arrow.get('1367900664123','x').format('YYYY-MM-DD HH:mm:ss:SSSSSS')" We can already see milliseconds and microseconds. I am not sure what kind of output format token 'x' will do.
I have tried this:

>>> import arrow
>>> arrow.get('1367900664123','x').format('x')
'123000'

this just shows the milliseconds and microseconds in the timestamp
but I'm not sure if it is what you want.

**Comment by jadchaar** [comment]
Hi @alstomli, my apologies. I forgot to include an example. The formatter currently works as follows:

```shell
venv ❯ python3
Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 16:52:21)
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import arrow
>>> arw = arrow.utcnow()
>>> arw.format("X")
'1575002210'
```

Currently, the "X" token returns an integer timestamp because it relies on the `timestamp` property of Arrow time objects, which has not been updated to return a float since it predates float timestamp support in datetime. @systemcatch we may need to update the timestamp property to return a float before this can be implemented (this may potentially be blocked by https://github.com/crsmithdev/arrow/issues/465). Or maybe there is an internal data structure on the Arrow object we can access to retrieve the seconds.

The "x" token would simply make a float UNIX timestamp like `1575002298.249394` (output of `time.time()`) into `1575002298249394` since this representation bakes the milliseconds (.249) and the microseconds (.000394) into an integer representation.

**Comment by alstomli** [comment]
got it

**Comment by alstomli** [comment]
got it

**Comment by alstomli** [comment]
Hi @jadchaar. When I use "x" token, if the utctime looks like '<Arrow [2019-11-29T22:22:57.154000+00:00]>' should the program outputs '1575066177154000' or '1575066177154' do we need to eliminate the zeros in the end?

**Comment by jadchaar** [comment]
Hmm @systemcatch what do you think? I am leaning on the side of not removing the trailing zeros for the sake of abstraction (e.g. multiply to get milli and micro regardless of whether there are zeros at the end).

**Comment by systemcatch** [comment]
> Hmm @systemcatch what do you think? I am leaning on the side of not removing the trailing zeros for the sake of abstraction (e.g. multiply to get milli and micro regardless of whether there are zeros at the end).

Hey @jadchaar I missed your comment. Errr I think it's probably best to leave the zeros in place.

One point to bear in mind is we are proposing a breaking change with the `X` formatting, we need to be careful doing this.

Current master;

```shell
(arrow) chris@ThinkPad:~/arrow$ python
Python 3.7.4 (default, Sep 19 2019, 11:01:37) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import arrow
>>> arw=arrow.utcnow()
>>> arw
<Arrow [2019-12-18T10:40:04.719476+00:00]>
>>> arw.format('X')
'1576665604'
```

In #737 

```shell
(arrow) chris@ThinkPad:~/arrow$ python
Python 3.7.4 (default, Sep 19 2019, 11:01:37) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import arrow
>>> arw=arrow.utcnow()
>>> arw
<Arrow [2019-12-18T11:05:50.204665+00:00]>
>>> arw.format('X')
'1576667150.204665'
```

**Comment by systemcatch** [comment]
Hey @jadchaar I've been working on implementing this. Currently the format will always default to microseconds, what do you think?

```shell
(arrow) chris@ThinkPad:~/arrow$ python
Python 3.7.4 (default, Sep 19 2019, 11:01:37) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import arrow
>>> arw=arrow.utcnow()
>>> arw.format("X")
'1585564976'
>>> arw.format("x")
'1585564976422134'
```

**Comment by jadchaar** [comment]
Looks good to me @systemcatch! You are keeping the 0s if there are any in the micro/milliseconds right?

**Comment by systemcatch** [comment]
Yeah they don't get cutoff.

```shell
(arrow) chris@ThinkPad:~/arrow$ python
Python 3.7.4 (default, Sep 19 2019, 11:01:37) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import arrow
>>> arw=arrow.utcnow()
>>> arw.format("x")
'1585663989425932'
>>> arw.floor('year').format("x")
'1577836800000000'
```

**Comment by jadchaar** [comment]
Looks good! PR those changes!

**Comment by systemcatch** [comment]
Implemented in #766
